lineage/crossval.py

```diff
@@ -0,0 +1,45 @@
+""" Cross validation. """
+import numpy as np
+from sklearn.utils import shuffle
+from copy import deepcopy
+import itertools as it
+
+def hide_observation(lineages: list, percentage: float) -> list:
+    """ Taking a list of lineages and the percentage of cells want to be masked, it marks those x% as -1.
+    This x% is selected from all cells and all lineages together such that we create a list of all cells in a population, hide x% and then regroup to their lineages.
+    We mark a random number of cells' observations as negative, to be removed from fitting.
+    :return new_lineages: list of lineages in which x% of cells were randomly masked. This forms our train lineages.
+    :return new_hide_index: list of arrays, each array corresponding to indexes of masked cells in each lineage.
+    :return obss: Those observations that have been masked, which are technically the test data.
+    """
+
+    new_lineages = deepcopy(lineages)
+    num_cells = 0
+    len_lineage = [] # remember the length of each lineage
+    for lin in lineages:
+        num_cells += len(lin.output_lineage)
+        len_lineage.append(len(lin.output_lineage))
+
+    # create the indexes for hidden observations
+    hide_index = np.zeros(num_cells)
+    hide_index[:int(num_cells*percentage)] = 1
+    hide_index = shuffle(hide_index)
```

Comment:
One more thought—this doesn't need to be an exact fraction. Why not just walk the cell tree, and then pull a Bernoulli random number for each cell and flip it if you get a 1? That further massively simplifies all of this.

Reply:
That certainly made it a lot smpler! Thank you!